azurerm/resource_arm_virtual_hub.go

```diff
@@ -0,0 +1,410 @@
+			}
+		}
+		if props.ExpressRouteGateway != nil {
+			if err := d.Set("express_route_gateway_id", props.ExpressRouteGateway.ID); err != nil {
+				return fmt.Errorf("Error setting `express_route_gateway_id`: %+v", err)
+			}
+		}
+		if err := d.Set("virtual_network_connection", flattenArmVirtualHubVirtualNetworkConnection(props.VirtualNetworkConnections)); err != nil {
+			return fmt.Errorf("Error setting `virtual_network_connection`: %+v", err)
+		}
+		if props.RouteTable != nil {
+			if err := d.Set("route", flattenArmVirtualHubRoute(props.RouteTable.Routes)); err != nil {
+				return fmt.Errorf("Error setting `route`: %+v", err)
+			}
+		}
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmVirtualHubDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Network.VirtualHubClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["virtualHubs"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting Virtual Hub %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for deleting Virtual Hub %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmVirtualHubVirtualNetworkConnection(input []interface{}) *[]network.HubVirtualNetworkConnection {
+	if len(input) == 0 {
+		return nil
+	}
+
+	results := make([]network.HubVirtualNetworkConnection, 0)
+
+	for _, item := range input {
+		v := item.(map[string]interface{})
+		name := v["name"].(string)
+		remoteVirtualNetworkId := v["remote_virtual_network_id"].(string)
+		allowHubToRemoteVnetTransit := v["allow_hub_to_remote_vnet_transit"].(bool)
+		allowRemoteVnetToUseHubVnetGateways := v["allow_remote_vnet_to_use_hub_vnet_gateways"].(bool)
+		enableInternetSecurity := v["enable_internet_security"].(bool)
+
+		result := network.HubVirtualNetworkConnection{
+			Name: utils.String(name),
+			HubVirtualNetworkConnectionProperties: &network.HubVirtualNetworkConnectionProperties{
+				RemoteVirtualNetwork: &network.SubResource{
+					ID: utils.String(remoteVirtualNetworkId),
+				},
+				AllowHubToRemoteVnetTransit:         utils.Bool(allowHubToRemoteVnetTransit),
+				AllowRemoteVnetToUseHubVnetGateways: utils.Bool(allowRemoteVnetToUseHubVnetGateways),
+				EnableInternetSecurity:              utils.Bool(enableInternetSecurity),
+			},
+		}
+
+		results = append(results, result)
+	}
+	return &results
+}
+
+func expandArmVirtualHubRoute(input []interface{}) *network.VirtualHubRouteTable {
+	if len(input) == 0 {
+		return nil
+	}
+
+	results := make([]network.VirtualHubRoute, 0)
+	for _, item := range input {
+		v := item.(map[string]interface{})
```

Comment:
We should check that this isn't nil before casting it

Reply:
Fixed